crates/driver/src/infra/config/cli.rs

```diff
@@ -1,13 +1,15 @@
 use {
-    std::{net::SocketAddr, path::PathBuf},
+    crate::domain::eth,
+    std::{path::PathBuf, str::FromStr},
     url::Url,
 };
 
 #[derive(Debug, clap::Parser)]
 pub struct Args {
-    /// The address to bind the driver to.
+    /// The address to bind the driver to. "auto" to bind to 0.0.0.0 and any
+    /// free port.
     #[clap(long, env)]
-    pub bind_addr: SocketAddr,
+    pub bind_addr: String,
```

Comment:
You can implement `api::Addr: Display + FromStr` and then:
```suggestion
    pub bind_addr: api::Addr,
```

Reply:
Ah - good point. NVM then.